### ISSUE TITLE
Exposing native store functions into public API

### ISSUE THREAD

**Issue by xor-gate** [ludocode/mpack#26]
Hi Nicholas,

I'm currently integrating mpack 0.7 and really like the `mpack_store|load*` functions to be used also elsewhere in my code. Only the `mpack_store_native_*_at` functions are note in the public header. This because I was smacked against the `warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]` when setting optimalisation to `-Os` for my firmware.

Is there any chance you can give them a place in the public reader API (header) ?

**Comment by ludocode** [comment]
I moved the store functions next to the load functions in `mpack-common.h` which is a much better place for them (especially since I will probably be using them in outside of the writer at some point.) Feel free to use them. Note that I also dropped the `_at` sufffix.

I might be optimizing them a bit more on ARM platforms at some point as well. See the discussion in #15 if you're interested.

**Comment by xor-gate** [comment]
Thanks for looking into it.

**Comment by ludocode** [comment]
FYI I've also dropped `_native` from the names in 5c4b0b1a2da6129cd0efeb2b8a15c30fd96050b4. Now they are just called e.g. `mpack_load_i32()`, `mpack_store_double()`, etc.

I wanted to keep something in the name to imply that they do endian conversion, but I kept forgetting what they were called. `_native` doesn't really imply that anyway, it just implies that it's outside of a reader/writer stream. `_network` or `_bigendian` might have been better but they are too ugly, so the `mpack_` prefix alone is good enough.
